### PR TITLE
Closes #3870: bug in reshape for bigint type

### LINKED ISSUE #3870: bug in reshape for bigint type
The `reshape` function gives the incorrect result for the `bigint` type, as shown in the following example:
```
In [3]: dtype = "bigint"

In [4]: size = 10

In [5]:         x = ak.arange(size, dtype=dtype).reshape((1, size, 1))

In [6]: x
Out[6]: array([array([array([0]) array([1]) array([2]) array([3]) array([4]) array([224251899497]) array([1919968626]) array([78401563153491]) array([127961796865894]) array([111052610564705])])])
```

### LINKED ISSUE #3908: flatten for bigint
Write a `flatten` procedure for `bigint` type, to mitigate bugs similar to https://github.com/Bears-R-Us/arkouda/issues/3870

### PR DESCRIPTION
**Pull request by ajpotts**
Fixes a bug where `reshape` function gives the incorrect result for the `bigint` type, as shown in the following example:
```
In [3]: dtype = "bigint"

In [4]: size = 10

In [5]:         x = ak.arange(size, dtype=dtype).reshape((1, size, 1))

In [6]: x
Out[6]: array([array([array([0]) array([1]) array([2]) array([3]) array([4]) array([224251899497]) array([1919968626]) array([78401563153491]) array([127961796865894]) array([111052610564705])])])
```
Fixes the issue by creating a separate `flatten` and `unflatten` function in AryUtil.chpl for the bigint case.

Closes #3870: bug in reshape for bigint type
Closes #3908: flatten for bigint

### CODE REVIEW

**Comment by ajpotts** [comment]
I believe the cause of the bug is this section the `unflatten` procedure:
```
            get(
              c_ptrTo(unflat[low]),
              getAddr(a[flatSlice.low]),
              locInStart,
              c_sizeof(t) * flatSlice.size
            );
```
I think the fact that `BigInt` does not have constant size, the data cannot be bulk copied using the address.

The fix was write a separate `unflatten` procedure for the bigint case.

File: src/AryUtil.chpl [hunk 1]
```diff
@@ -1044,7 +1080,7 @@ module AryUtil
       // index -> order for the input array's indices
       // e.g., order = k + (nz * j) + (nz * ny * i)
       inline proc indexToOrder(idx: rank*?t): t
-      where (t==int) || (t==uint(64)) {
+        where (t==int) || (t==uint(64)) {
```
Comment (jaketrookman) [inline src/AryUtil.chpl:1083]: I think you might want to fix the alignment below this too, unless the linter is giving you trouble

Comment (jaketrookman) [reply src/AryUtil.chpl:1083]: I was specifically refering to the indentation below this of what is inside the `where`

Comment (ajpotts) [reply src/AryUtil.chpl:1083]: I didn't  write that but I think we should have a team discussion about what indentation we should use.  I see throughout the code very commonly that the where statement is indented to the same levels as the code.

Comment (jaketrookman) [reply src/AryUtil.chpl:1083]: Discussed and agreed on format in our Tuesday meeting.

**Review by drculhane — APPROVED** [review]
Looks good.  You'll want to rename test_reshape_bug_reproducer :)

**Review by jaketrookman — APPROVED** [review]
Reapproving after formatting changes

**Pull request merged**